packages/gatsby/src/redux/reducers/__tests__/page-data-dependencies.ts

```diff
@@ -95,11 +97,13 @@ describe(`add page data dependency`, () => {
     // Add different action
     state = reducer(state, action2)
 
-    expect(state.connections.get(`MarkdownRemark`).size).toEqual(2)
-    expect(state.nodes.get(1).size).toEqual(2)
+    expect(
+      (state.connections.get(`MarkdownRemark`) as Set<string>).size
```

Comment:
If casting here is to avoid potential `undefined` let's use non-null assertion operator
```suggestion
      state.connections.get(`MarkdownRemark`)!.size
```

if it it's not about that - then why is this needed?

Similarly line below (+ question why is this `number | string` in first place - but there is other comment closer to typing source about this

Reply:
Actually - let's use optional chaining which is much safer and still will work for this use case
```suggestion
      state.connections.get(`MarkdownRemark`)?.size
```